src/main.rs

```diff
@@ -256,171 +327,42 @@ impl<T: Read + Write + imap::extensions::idle::SetReadTimeout> Connection<T> {
-                        password,
-                        notification_command: t.get("notificationcmd").and_then(
-                            |raw_v| match raw_v.as_str() {
-                                Some(v) => Some(v.to_string()),
-                                None => parse_failed("notificationcmd", "string"),
-                            },
-                        ),
-                        folders: {
-                            // Parse the folders field
-                            let mut folders: Vec<String> = t
-                                .get("folders")
-                                .and_then(|raw_v| {
-                                    raw_v
-                                        .as_array()
-                                        .map(|v| {
-                                            v.iter()
-                                                .filter_map(|raw_v| {
-                                                    raw_v
-                                                        .as_str()
-                                                        .map(|v| v.to_string())
-                                                        .or_else(|| parse_failed("folders", "str"))
-                                                })
-                                                .collect()
-                                        })
-                                        .or_else(|| parse_failed("folders", "array"))
-                                })
-                                .unwrap_or_default();
-
-                            // Parse the old folder field and push it to the list of folders
-                            if let Some(folder) = t.get("folder").and_then(|raw_v| {
-                                raw_v
-                                    .as_str()
-                                    .map(|x| x.to_string())
-                                    .or_else(|| parse_failed("folder", "string"))
-                            }) {
-                                folders.push(folder);
-                            }
-
-                            if folders.is_empty() {
-                                vec![String::from("INBOX")]
-                            } else {
-                                folders
-                            }
-                        },
-                    })
-                }
-            })
-            .collect(),
-        None => {
-            println!("Could not parse configuration file buzz.toml: not a table");
-            return;
-        }
-    };
-
-    if accounts.is_empty() {
-        println!("No accounts in config; exiting...");
-        return;
+    let config = Config::from_file(
+        ProjectDirs::from("", "", "buzz")
+            .expect("Could not find valid home directory.")
+            .config_dir()
+            .with_file_name("buzz.toml"),
+    )?;
+
+    if config.accounts.is_empty() {
+        anyhow::bail!("No accounts in config; exiting...");
```

Comment:
I don't think this should be an `anyhow::bail!` — let's just print + return here (like we used to).